src/model_io/urdf/tests/PredictSensorsMeasurementUnitTest.cpp

```diff
@@ -0,0 +1,146 @@
+#include <cassert>
+#include <cstdio>
+#include <cstdlib>
+using namespace iDynTree;
+
+bool init(std::string fileName, Model &model, Traversal &traversal, SensorsList &sensorsList, SensorsMeasurements &predictedMeasurement)
+{
+    // load URDF model
+    bool ok = modelFromURDF(fileName,model);
+    ASSERT_EQUAL_DOUBLE(ok,true);
+    std::cout<<"Model "<<fileName.c_str()<<" created with :"<<model.getNrOfDOFs()<<" DoFs"<<std::endl;
+
+    ASSERT_EQUAL_DOUBLE(model.getNrOfLinks(),2);
+    ASSERT_EQUAL_DOUBLE(model.getNrOfJoints(),1);
+    ASSERT_EQUAL_DOUBLE(model.getNrOfDOFs(),1);
+    ASSERT_EQUAL_DOUBLE(model.getNrOfFrames(),3);
+    ASSERT_EQUAL_STRING(model.getLinkName(model.getDefaultBaseLink()),"link1");
+   
+    ok = model.computeFullTreeTraversal(traversal);
+
+    ASSERT_EQUAL_DOUBLE(ok,true);
+    
+    // Load sensorList
+    genericSensorsListFromURDF(fileName,sensorsList);
+
+    ASSERT_EQUAL_DOUBLE(sensorsList.getNrOfSensors(ACCELEROMETER),2);
+    ASSERT_EQUAL_DOUBLE(sensorsList.getNrOfSensors(GYROSCOPE),1);
+    
+    predictedMeasurement.resize(sensorsList);
+}
+
+bool runTest(const int& expID,const Model& model,const Traversal& traversal, const SensorsList& sensorsList, SensorsMeasurements& predictedMeasurement)
+{     
+    // quantities to be set according to experiment
+    FreeFloatingPos robotPos(model);
+    FreeFloatingVel robotVel(model);
+    FreeFloatingAcc robotAcc(model);
+    LinAcceleration gravity(0,0,0);
+    
+    iDynTree::LinkPositions linkPos(model);
+    iDynTree::LinkVelArray linkVel(model);
+    iDynTree::LinkAccArray linkAcc(model);
+    
+    PredictSensorsMeasurements predictedSensors;
+    LinAcceleration accl1(0,0,0),accl2(0,0,0);
+    AngVelocity gyro1(0,0,0);
+    
+    
+    //experiments 1-accelerometer gravity test, 2-angularVelocity test, 3-angularAccelerationTest 
+    
+    std::cout<<"------------------------\n";
+    std::cout<<"Experiment "<<expID<<"\n";
+    switch(expID)
+    {
+        case 1 :gravity= LinearMotionVector3(0,0,9.8);
+                robotAcc.baseAcc() = SpatialAcc::Zero();
+                break;
+            
+        case 2 :gravity= LinearMotionVector3(0,0,0);
+                robotAcc.baseAcc() = SpatialAcc::Zero();
+                robotVel.jointVel()(0) = gyroTestVal;
+                break;
+        case 3 :gravity= LinearMotionVector3(0,0,0);
+                robotAcc.baseAcc() = SpatialAcc::Zero();
+                robotVel.jointVel()(0) = 0;          
+                robotAcc.jointAcc()(0) = acclTestVal;
+                break;
+    }
+    
+    predictedSensors.makePrediction(model,traversal,robotPos,robotVel,robotAcc,linkPos,linkVel,linkAcc,gravity,sensorsList,predictedMeasurement);
+    predictedMeasurement.getMeasurement(ACCELEROMETER,0,accl1);
+    predictedMeasurement.getMeasurement(ACCELEROMETER,1,accl2);
+    predictedMeasurement.getMeasurement(GYROSCOPE,0,gyro1);
+    std::cout<<"Predicted Measurement (accl1): " <<accl1.toString()<<"\n";
```

Comment:
@traversaro for some reason this doesnt trigger a memcheck error if I use reservedToString() instead of toString()

Even though those functions are identical